x-pack/plugins/security/server/session_management/session_index.ts

```diff
@@ -353,26 +340,57 @@ export class SessionIndex {
     this.options.logger.debug(`Running cleanup routine.`);
 
     const now = Date.now();
+    const providersSessionConfig = this.options.config.authc.sortedProviders.map((provider) => {
+      return {
+        boolQuery: {
+          bool: {
+            must: [
+              { term: { 'provider.type': provider.type } },
+              { term: { 'provider.name': provider.name } },
+            ],
+          },
+        },
+        ...this.options.config.session.getExpirationTimeouts(provider),
+      };
+    });
 
     // Always try to delete sessions with expired lifespan (even if it's not configured right now).
     const deleteQueries: object[] = [{ range: { lifespanExpiration: { lte: now } } }];
 
-    // If lifespan is configured we should remove any sessions that were created without one.
-    if (this.options.config.session.lifespan) {
-      deleteQueries.push({ bool: { must_not: { exists: { field: 'lifespanExpiration' } } } });
-    }
+    // If session belongs to a not configured provider we should also remove it.
+    deleteQueries.push({
+      bool: {
+        must_not: { bool: { should: providersSessionConfig.map(({ boolQuery }) => boolQuery) } },
```

Comment:
nit: I know it's the default value, but would you be ok adding a `minimum_should_match: 1` to this? I think it would make the query easier for me to parse (personally). Having explicit `minimum_should_match` takes the guess work out of what our intent is here.

Reply:
Sure, that wouldn't hurt.